perma_web/perma/tests/test_views_auth.py

```diff
@@ -8,152 +7,145 @@
+                                 {'username': username,
+                                  'password': password})
+    if expect_success:
+        assert response.status_code == 302
+        assert 'login' not in response['Location']
+        assert '_auth_user_id' in perma_client.session
+    else:
+        assert '_auth_user_id' not in perma_client.session
+    return response
+
+def test_login(perma_client, link_user):
+    """
+    Test the login form
+    We should get redirected to the create page
+    """
+    # Login through our form and make sure we get redirected to our create page,
+    # no matter how the email address is capitalized
+    assert LinkUser.objects.filter(email__iexact=link_user.email).count() == 1
+
+    attempt_login(perma_client, link_user.email, TEST_USER_PASSWORD)
+    perma_client.logout()
+    attempt_login(perma_client, link_user.email.upper(), TEST_USER_PASSWORD)
+    perma_client.logout()
+    attempt_login(perma_client, link_user.email.title(), TEST_USER_PASSWORD)
+    perma_client.logout()
+    attempt_login(perma_client, randomize_capitalization(link_user.email), TEST_USER_PASSWORD)
+
+def test_deactived_user_login(perma_client, deactivated_user):
+    submit_form(perma_client, 'user_management_limited_login',
+                    data = {'username': deactivated_user.email,
+                            'password': TEST_USER_PASSWORD},
+                    success_url=reverse('user_management_account_is_deactivated'))
+    assert '_auth_user_id' not in perma_client.session
+
+def test_unactived_user_login(perma_client, unactivated_user):
+    submit_form(perma_client, 'user_management_limited_login',
+                        data = {'username': unactivated_user.email,
+                                'password': 'pass'},
+                        success_url=reverse('user_management_not_active'))
+    assert '_auth_user_id' not in perma_client.session
+
+def test_logout(perma_client, link_user):
+    """
+    Test our logout link
+    """
+
+    # Login with our client and logout with our view
+    attempt_login(perma_client, link_user.email, TEST_USER_PASSWORD)
+    assert '_auth_user_id' in perma_client.session
+    perma_client.get(reverse('logout'))
+    submit_form(perma_client, 'logout')
+    assert '_auth_user_id' not in perma_client.session
+
+def test_password_change(perma_client, link_user):
+    """
+    Let's make sure we can login and change our password
+    """
+
+    attempt_login(perma_client, link_user.email, TEST_USER_PASSWORD)
+    assert '_auth_user_id' in perma_client.session
+
+    perma_client.post(reverse('password_change'),
+        {'old_password':TEST_USER_PASSWORD, 'new_password1':'Changed-password1',
+        'new_password2':'Changed-password1'})
+
+    perma_client.logout()
+
+    # Try to login with our old password
+    attempt_login(perma_client, link_user.email, TEST_USER_PASSWORD, expect_success=False)
+    assert '_auth_user_id' not in perma_client.session
```

Comment:
Same here: because of `expect_success=False`, I think that assertion has already been made.